src/app/item-page/simple/field-components/specific-field/metrics/altmetric/item-page-altmetric-field.directive.ts

```diff
@@ -0,0 +1,83 @@
+import { Directive, ElementRef, Input, OnInit, Renderer2 } from '@angular/core';
+import { Item } from 'src/app/core/shared/item.model';
+
+/**
+ * This directive adds the data-* attribute for the Altmetric badge dependening on the first identifier found in the item
+ */
+@Directive({
+  selector: '[dsAltmetricData]',
+})
+export class AltmetricDirective implements OnInit {
+  @Input() item: Item;
+
+  constructor(private renderer: Renderer2, private elementRef: ElementRef) {}
+
+  ngOnInit(): void {
+    const identifier = this.obtainFirstValidID(this.initItemIdentifiers());
+    if (identifier !== undefined) {
+      this.renderer.setAttribute(
+        this.elementRef.nativeElement,
+        identifier.name,
+        this.applyRegex(identifier.value, identifier.regex)
+      );
+    }
+  }
+
+  /**
+   * This initialize an array of identifiers founded in the item.
+   * It search for DOI, Handle, PMID, ISBN, ARXIV and URI.
+   * Some identifiers may be stored with more text than the ID so this objects has a regex property to clean it
+   */
+  private initItemIdentifiers(): any[] {
+    return [
+      {
+        name: 'data-doi',
+        value: this.item.firstMetadataValue('dc.identifier.doi'),
```

Comment:
I don't believe this is correct.  Currently DSpace is storing DOIs in `dc.identifier.uri` and/or `dc.identifier` based on this ticket https://github.com/DSpace/DSpace/issues/5565

Maybe we need a way to check for DOIs in all three of these fields?  I'm OK with the first check being `dc.identifier.doi`...but if not found there, we may want to check other fields for DOIs too.